domain/port/service/service.go

```diff
@@ -26,16 +29,23 @@ type AtomicState interface {
 	// net-node.
 	GetColocatedOpenedPorts(ctx domain.AtomicContext, unitUUID string) ([]network.PortRange, error)
 
-	// GetOpenedEndpointPorts returns the opened ports for a given endpoint of a
-	// given unit.
-	GetOpenedEndpointPorts(ctx domain.AtomicContext, unitUUID, endpoint string) ([]network.PortRange, error)
+	// GetUnitOpenedPortsUUID returns the opened ports for the given unit with the
+	// UUID of the port range. The opened ports are grouped by endpoint.
+	GetUnitOpenedPortsUUID(ctx domain.AtomicContext, unitUUID string) (map[string][]port.PortRangeUUID, error)
 
 	// GetEndpoints returns all endpoints for a given unit.
-	GetEndpoints(ctx domain.AtomicContext, unitUUID string) ([]string, error)
+	GetEndpoints(ctx domain.AtomicContext, unitUUID string) ([]port.Endpoint, error)
+
+	// AddEndpoints adds the endpoints to a given unit. Return the added endpoints
+	// with their corresponding UUIDs.
+	AddEndpoints(ctx domain.AtomicContext, unitUUID string, endpoints []string) ([]port.Endpoint, error)
 
-	// UpdateUnitPorts opens and closes ports for the endpoints of a given unit.
-	// The opened and closed ports for the same endpoints must not conflict.
-	UpdateUnitPorts(ctx domain.AtomicContext, unitUUID string, openPorts, closePorts network.GroupedPortRanges) error
+	// AddOpenedPorts adds the given port ranges to the database. Port ranges must
+	// be grouped by endpoint UUID.
+	AddOpenedPorts(ctx domain.AtomicContext, portRangesByEndpointUUID network.GroupedPortRanges) error
```

Comment:
Don't pass core types to state. The types should be serialised to domain level values. These values can live in `domain/port/types.go`.

Reply:
Ignoring for now after chatting with @SimonRichardson. These types should be moved out of core into `domain/ports/types.go`, but doesn't need to be changed now